dag-syntax.cddl

```diff
@@ -0,0 +1,24 @@
+application = {
+    applicationName: tstr,
+    nodes: [ * node ],
+}
+
+node = {
+    task: tstr,
+    name: tstr,
+    type: "producer" / "prosumer" / "consumer",
+    description: tstr,
+    executor: "container",
+    container: {
+        image: tstr,
+        resources: {
+            cpu: uint,
+            memory: tstr, ; TODO: Discuss the unit we want to use here
+            taskComplexity: uint,
```

Comment:
I think that `taskComplexity` is probably irrelevant for the container executor and should probably belong in a different section.
Maybe it would make sense to have a separate section for information that is relevant to the scaling algorithm?

Reply:
Yeah, that sounds good to me 👍

Reply:
I introduced a `scalingParameters` field now that contains `taskComplexity` as a sub-field. The other suggestions/comments should now also be addressed, and the example DAG JSON file updated accordingly.